lib/filewatch/tail.rb

```diff
@@ -11,13 +11,11 @@ class Tail
     attr_writer :target
 
     def self.new_observing(opts = {})
-      new.tap do |instance|
-        instance.target = ObservingTail.new(opts)
-      end
+      new({}, ObservingTail.new(opts))
     end
 
-    def initialize(opts = {})
-      @target = YieldingTail.new(opts)
+    def initialize(opts = {}, target = YieldingTail.new(opts))
```

Comment:
It does - I agree sort of, more so if one uses the hash shortcut:
```ruby
FileWatch::Tail.new(:sincedb_path =>"",
  YieldingTail.new(
    :sincedb_path => sincedb_path, :start_new_files_at => :beginning, :stat_interval => 0
  )
)
```
but the switching of the arguments would break:
- the use of opts set in the previous argument when the default value for target is evaluated because target is not supplied,
- the constructor expectation set already,

In practice no one should know about this dependency injection argument, but if someone want to write a different kind of tailing class - they have an injection mechanism.